### PR TITLE
Document gh2gcs use case and SIG Release managed buckets

### PR DESCRIPTION
**Pull request by xmudrii**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Document gh2gcs use case and SIG Release managed buckets

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

/assign @saschagrunert @justaugustus

### CODE REVIEW

**Review by cpanato — APPROVED** [review]
/lgtm
thanks for this

File: cmd/gh2gcs/README.md [hunk 1]
```diff
@@ -34,3 +34,16 @@ INFO Writing assets to /tmp/test/kubernetes/kubernetes/v1.18.0
...
+
+## Use case
+
+GitHub has a rate limit on downloading release artifacts. The artifacts that are often used (e.g. in CI) are uploaded to Google Cloud Storage, as GCS doesn't have read rate limits. This improves test stability as flakes related to hitting rate limits are avoided.
+
+## SIG Release managed buckets
+
+The following GCS buckets are managed by SIG Release:
+
+- k8s-artifacts-cni - contains CNI plugins artifacts
```
Comment (saschagrunert) [inline cmd/gh2gcs/README.md:46]: Let's link the CNI plugins repo here, too.

File: cmd/gh2gcs/README.md [hunk 2]
```diff
@@ -34,3 +34,16 @@ INFO Writing assets to /tmp/test/kubernetes/kubernetes/v1.18.0
...
+- k8s-artifacts-cri-tools - contains [CRI tools](https://github.com/kubernetes-sigs/cri-tools) artifacts (`crictl` and `critest`)
+
+The artifacts are pushed to GCS by [Release Managers](https://github.com/kubernetes/sig-release/blob/master/release-managers.md). The pushing is done manually by running the appropriate `gh2gcs` command.
```
Comment (saschagrunert) [inline cmd/gh2gcs/README.md:49]: Can we add an example for those commands?

**Review by saschagrunert — COMMENTED** [review]
Thanks, just two suggestions from my side :point_down:

File: cmd/gh2gcs/README.md (hunk 2 above)
Comment (xmudrii) [reply cmd/gh2gcs/README.md:49]: We have an example above, in the Usage part. Should I add another or is it enough?

Comment (saschagrunert) [reply cmd/gh2gcs/README.md:49]: Oops, did not catch that. Yep that should be fine.

File: cmd/gh2gcs/README.md (hunk 1 above)
Comment (xmudrii) [reply cmd/gh2gcs/README.md:46]: @saschagrunert Do we have our repo or linking to https://github.com/containernetworking/plugins is okay?

Comment (saschagrunert) [reply cmd/gh2gcs/README.md:46]: I think linking to https://github.com/containernetworking/plugins should be fine :) We could also recommend to the manager to watch those repositories for possible new releases.

Comment (xmudrii) [reply cmd/gh2gcs/README.md]: @saschagrunert Fixed, PTAL.

**Review by saschagrunert — APPROVED** [review]
/lgtm
/approve

**Pull request merged**